From f119fbe347c8920f108d1ff152d5dc1bc0143fab Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:05:06 -0400
Subject: [fs] btrfs: fix btrfs_print_leaf for skinny metadata

Message-id: <1407179109-13714-420-git-send-email-sandeen@redhat.com>
Patchwork-id: 87375
O-Subject: [RHEL7.1 PATCH 419/422] Btrfs: fix btrfs_print_leaf for skinny metadata
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Josef Bacik <jbacik@fb.com>

commit be2c765dff9e5584965f78853c2addd2bb926946
Author: Josef Bacik <jbacik@fb.com>
Date:   Wed Jul 2 10:20:48 2014 -0700

    Btrfs: fix btrfs_print_leaf for skinny metadata

    We wouldn't actuall print the extent information if we had a skinny metadata
    item, this fixes that.  Thanks,

    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/print-tree.c b/fs/btrfs/print-tree.c
index 6efd70d..9626b4a 100644
--- a/fs/btrfs/print-tree.c
+++ b/fs/btrfs/print-tree.c
@@ -54,7 +54,7 @@ static void print_extent_data_ref(struct extent_buffer *eb,
         btrfs_extent_data_ref_count(eb, ref));
 }
 
-static void print_extent_item(struct extent_buffer *eb, int slot)
+static void print_extent_item(struct extent_buffer *eb, int slot, int type)
 {
  struct btrfs_extent_item *ei;
  struct btrfs_extent_inline_ref *iref;
@@ -63,7 +63,6 @@ static void print_extent_item(struct extent_buffer *eb, int slot)
  struct btrfs_disk_key key;
  unsigned long end;
  unsigned long ptr;
- int type;
  u32 item_size = btrfs_item_size_nr(eb, slot);
  u64 flags;
  u64 offset;
@@ -88,7 +87,8 @@ static void print_extent_item(struct extent_buffer *eb, int slot)
         btrfs_extent_refs(eb, ei), btrfs_extent_generation(eb, ei),
         flags);
 
- if (flags & BTRFS_EXTENT_FLAG_TREE_BLOCK) {
+ if ((type == BTRFS_EXTENT_ITEM_KEY) &&
+     flags & BTRFS_EXTENT_FLAG_TREE_BLOCK) {
   struct btrfs_tree_block_info *info;
   info = (struct btrfs_tree_block_info *)(ei + 1);
   btrfs_tree_block_key(eb, info, &key);
@@ -223,7 +223,8 @@ void btrfs_print_leaf(struct btrfs_root *root, struct extent_buffer *l)
     btrfs_disk_root_refs(l, ri));
    break;
   case BTRFS_EXTENT_ITEM_KEY:
-   print_extent_item(l, i);
+  case BTRFS_METADATA_ITEM_KEY:
+   print_extent_item(l, i, type);
    break;
   case BTRFS_TREE_BLOCK_REF_KEY:
    printk(KERN_INFO "\t\ttree block backref\n");
-- 
1.7.1